### DISCUSSION TITLE
How can I pass the window list from Python to rofi?

### DISCUSSION THREAD

**Discussion by s-aleck** [davatorium/rofi#1469] [Q&A] [answered]
The  script (which draws from various sources available online, including Stack Overflow):

```
import subprocess
import re

def sorting(a_list):
    master_sorting_list = []
    for sublist in a_list:
        sorting_list = []
        for member in sublist:
            for character in member.casefold():
                sorting_list.append(ord(character))
            master_sorting_list.append(sorting_list)
    return(master_sorting_list)

command = "wmctrl -lpx"
my_string = subprocess.check_output(["/bin/bash", "-c", command]).decode("utf-8")
my_list = my_string.split("\n")
main_list = []
for row in my_list:
    if row != "":
        elements = re.split("\s+", row, 5)
        elements.pop(0)
        elements.pop(1)
        elements.pop(2)
        elements[0] = " ".join(("Desktop", elements[0]))
        elements[1] = elements[1].split(".")[1]
        if elements[0] != "Desktop -1":
            main_list.append(elements)
main_list.sort(key=sorting)
for l in main_list:
    print(l)
```

The mode Specs specify that my script has to write choices to `stdout` separated by `\n`. 

I run:

```
rofi -modi yourscript:/home/path ....py -show yourscript
```

I get a `rofi` window, but without choices. How can I pass the choices to `rofi`?

SOLVED: Thanks to DaveDavenport's advice, I added the shebang:

```
#!/usr/bin/env python
```

Now it works.

**Comment by DaveDavenport** [comment, marked as answer]
that python file does not work on python2 or python3. it sais subprocess is unknown.

**Reply by s-aleck** [reply]
Sorry. I started the code bit with:

"```import subprocess"

and the line got left out. I edited my first post.

I forgot to explicitly state that I'm running Linux.

**Reply by DaveDavenport** [reply]
it works fine for me..  did you make the script executable and give it the right [shebang](https://en.wikipedia.org/wiki/Shebang_(Unix))?
